examples/3d/motion_blur.rs

```diff
@@ -0,0 +1,401 @@
+            let [x, z] = pos.into();
+            commands.spawn(PbrBundle {
+                mesh: sphere.clone(),
+                material: leaves.clone(),
+                transform: Transform::from_xyz(x, 0.0, z).with_scale(Vec3::splat(0.4)),
+                ..default()
+            });
+            commands.spawn(PbrBundle {
+                mesh: capsule.clone(),
+                material: trunk.clone(),
+                transform: Transform::from_xyz(x, -0.4, z).with_scale(Vec3::new(0.05, 0.5, 0.05)),
+                ..default()
+            });
+        }
+    };
+    spawn_with_offset(0.07);
+    spawn_with_offset(-0.07);
+}
+
+fn setup_ui(mut commands: Commands) {
+    let style = TextStyle {
+        font_size: 24.0,
+        ..default()
+    };
+    commands.spawn(
+        TextBundle::from_sections(vec![
+            TextSection::new(String::new(), style.clone()),
+            TextSection::new(String::new(), style.clone()),
+            TextSection::new("1/2: -/+ shutter angle (blur amount)\n", style.clone()),
+            TextSection::new("3/4: -/+ sample count (blur quality)\n", style.clone()),
+            TextSection::new("Spacebar: cycle camera\n", style.clone()),
+        ])
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            top: Val::Px(12.0),
+            left: Val::Px(12.0),
+            ..default()
+        }),
+    );
+}
+
+fn keyboard_inputs(
+    mut settings: Query<&mut MotionBlur>,
+    presses: Res<ButtonInput<KeyCode>>,
+    mut text: Query<&mut Text>,
+    mut camera: ResMut<CameraMode>,
+) {
+    let mut settings = settings.single_mut();
+    if presses.just_pressed(KeyCode::Digit1) {
+        settings.shutter_angle -= 0.25;
+    } else if presses.just_pressed(KeyCode::Digit2) {
+        settings.shutter_angle += 0.25;
+    } else if presses.just_pressed(KeyCode::Digit3) {
+        settings.samples = settings.samples.saturating_sub(1);
+    } else if presses.just_pressed(KeyCode::Digit4) {
+        settings.samples += 1;
+    } else if presses.just_pressed(KeyCode::Space) {
+        *camera = match *camera {
+            CameraMode::Track => CameraMode::Chase,
+            CameraMode::Chase => CameraMode::Track,
+        };
+    }
+    settings.shutter_angle = settings.shutter_angle.clamp(0.0, 1.0);
+    settings.samples = settings.samples.clamp(0, 64);
+    let mut text = text.single_mut();
+    text.sections[0].value = format!("Shutter angle: {:.2}\n", settings.shutter_angle);
+    text.sections[1].value = format!("Samples: {:.5}\n", settings.samples);
+}
+
+// Parametric function for a looping race track. `offset` will return the point offset normal to the center of the race course.
+fn race_track_shape(offset: f32, t: f32) -> Vec2 {
+    let x_tweak = 2.0;
+    let y_tweak = 3.0;
+    let scale = 8.0;
+    let x0 = (x_tweak * t).sin();
+    let y0 = (y_tweak * t).cos();
+    let dx = x_tweak * (x_tweak * t).cos();
+    let dy = y_tweak * -(y_tweak * t).sin();
```

Comment:
I think this would be clearer as:

```rust
let dt = vec2(x_tweak * (x_tweak * t).cos(), y_tweak * (-y_tweak * t).sin()).normalize_or_zero();
vec2(x0 + offset, y0 - offset) * dt.yx() * scale
```

Up to you.

Reply:
This ended up breaking things, so I kept it unchagned.